### PR TITLE
[TSC OSTK-Release] Create selection criteria

### LINKED ISSUE #1498: [TSC OSTK-Release] OpenStack Release Selection Criteria
Create Criteria document based on the 22 April 2020 vF2F discussions

### PR DESCRIPTION
**Pull request by pgoyal01**
Fixes #1498 

Incorporates content discussed at vF2F on 22nd April 2020.

### CODE REVIEW

**Comment by pgoyal01** [comment]
@walterkozlowski Yes. Sentence changed to:

When a new CNTT OpenStack baseline release is selected, it will trigger the creation of a new reelase of the Reference Architecture (RA-1), Reference Implementation (RI-1) and Reference Conformance (RC-1) documents.

Thanks

**Comment by ulikleber** [comment]
Is it intentional to have also the Edge use case part of this PR?

**Comment by kedmison** [comment]
Are the Edge use cases additions to doc/tech/usecases.md intended to be included in this PR?  They seem somewhat orthogonal to the PR description.

File: doc/tech/openstack-release.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+-	criteria for the triggering of the next baseline selection
+-	criteria to use in choosing the next OpenStack release, and 
+-	the number of OpenStack releases to be supported by CNTT
+
+When a new CNTT OpenStack baseline release is selected, it will trigger the creation of a new reelase of the Reference Architecture (RA-1), Reference Implementation (RI-1) and Reference Conformance (RC-1) documents. Please note that while a new OpenStack release selection may only trigger updates to certain sections, all document releases will be complete and can be utilised on their own independent of previous releases.
+
+## 1.1 Triggering Events for next release selection
+This section specifies events that may trigger the selection of the next OpenStack release.
+-	Complete change in architecture: OpenStack or OpenStack Service 
+-	New OpenStack features needed by CNTT workloads
```
[the remark this answers is not in the document]
Comment (pgoyal01) [reply doc/tech/openstack-release.md:12]: @ASawwaf done - thanks

File: doc/tech/openstack-release.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+-	Security Fix (not fixed through a patch; OpenStack or OS)
+-	Current CNTT [OpenStack release]( https://releases.openstack.org/) entered “[Extended Maintenance]( https://docs.openstack.org/project-team-guide/stable-branches.html#maintenance-phases)” phase approximately 18 months ago
```
[the remark this answers is not in the document]
Comment (pgoyal01) [reply doc/tech/openstack-release.md:14]: @ASawwaf Done - Thanks

File: doc/tech/openstack-release.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+## 1.3 OpenStack Release Selection Criteria
+The OpenStack Release Selection Committee shall utilize the following criteria, and any other criteria that it unanimously agrees to, in selecting the next CNTT OpenStack baseline release:
+-	The latest OpenStack release that was released approximately 6 months ago
+    - The Committee may agree to relax or extend the 6 months period based on it knowledge of OpenStack releases
+-	The OpenStack release should be supported by the OPNFV Installer (Airship)
+-	Backward Compatibility: ensure API support
+-	Consider OpenStack Distribution vendors and their extended support versions
+
+## 1.4 Deprecation of CNTT OpenStack Releases
+CNTT shall support no more than 2 (two) OpenStack releases at any given time. Thus, on selection of a new CNN OpenStack baseline release, an existing CNTT OpenStack release may be deprecated. The selection of new CNTT OpenStack release n, triggers the deprecation of the n-2 release. On the completion of the Reference Architecture for release n, the release n-2 will stand deprecated.
```
[the remark this answers is not in the document]
Comment (pgoyal01) [reply doc/tech/openstack-release.md:30]: @ASawwaf Done - Thanks

File: doc/tech/openstack-release.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+The committee may agree by unanimous agreement to adjust the OpenStack Release Selection Criteria.
+
+## 1.3 OpenStack Release Selection Criteria
+The OpenStack Release Selection Committee shall utilize the following criteria, and any other criteria that it unanimously agrees to, in selecting the next CNTT OpenStack baseline release:
+-	The latest OpenStack release that was released approximately 6 months ago
+    - The Committee may agree to relax or extend the 6 months period based on it knowledge of OpenStack releases
+-	The OpenStack release should be supported by the OPNFV Installer (Airship)
+-	Backward Compatibility: ensure API support
+-	Consider OpenStack Distribution vendors and their extended support versions
+
```
[the remark this answers is not in the document]
Comment (pgoyal01) [reply doc/tech/openstack-release.md:28]: @ASawwaf it is more than study -- the team may consider the availability of an extended release version from a vendor in making their choice as it kind of guarantees availability of fixes for key issues.

**Comment by rabi-abdel** [comment]
@pgoyal01 resolved, remove use cases section from it.

**Comment by pgoyal01** [comment]
@karinesevilla Thanks. Incorporated your suggested rewrite. 

W.r.t. APIs -- most new OSTK release have at least some API changes -- may not be significant.

W.r.t. OPNFV installer dependency, we need some other opinions | this was an ask

**Comment by pgoyal01** [comment]
@rabi-abdel deleted usecases. Thanks

**Comment by pgoyal01** [comment]
@ulikleber @kedmison @ASawwaf @walterkozlowski deleted usecase.md file from the patch. Thanks

**Comment by rabi-abdel** [comment]
@karinesevilla can you please re-review/approve?

File: doc/tech/openstack-release.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+## 1.2 OpenStack Release Selection Committee
+On the occurrence of any of the triggering events, the TSC shall constitute an OpenStack Release Selection Committee composed of a maximum of 7 (seven) members representing both operators and vendors. These committee members shall be from active CNTT member organisations and meet the criteria specified for “[Voting Representatives]( https://github.com/cntt-n/CNTT/blob/master/doc/gov/chapters/chapter05.md#52-voting-representatives)”. 
+The committee decisions shall be by consensus and no decision shall be made without at least 5 members agreeing.
+The committee may agree by unanimous agreement to adjust the OpenStack Release Selection Criteria.
+
+## 1.3 OpenStack Release Selection Criteria
+The OpenStack Release Selection Committee shall utilize the following criteria, and any other criteria that it unanimously agrees to, in selecting the next CNTT OpenStack baseline release:
+-	The latest OpenStack release that was released approximately 6 months ago
+    - The Committee may agree to relax or extend the 6 months period based on it knowledge of OpenStack releases
```
Comment (kedmison) [inline doc/tech/openstack-release.md:24]: Minor nit.  Approving the document, either way.
```suggestion
    - The Committee may agree to relax or extend the 6 months period based on its knowledge of OpenStack releases
```

**Pull request merged**